inst/extdata/install-packages.R

```diff
@@ -75,47 +72,45 @@ displayProgress <- function(current, success = TRUE, message = NULL, suppressOut
 
 #' Install all osps packages and their dependencies.
 #'
-#' @param rtoolsPath Path to where rtools are installed. If `NULL` (default),
-#' the path is deduced from system environment variables.
 #' @param rclrVersion Version of rClr package. Default is 0.9.2 for Windows R4.
 #' @param developerVersion If `FALSE` (default), release verions of the packages
 #' will be installed. If `TRUE`, latest developer builds of the osps packages
 #' will be installed
+#' @param lib character vector giving the library directories where to install
+#' the packages. Recycled as needed. If missing, defaults to the first element of .libPaths().
 #' @param suppressOutput
 #'
 #' @return
 #' @export
 #'
 #' @examples
-installOSPPackages <- function(rtoolsPath = NULL, rclrVersion = "0.9.2",
+installOSPPackages <- function(rclrVersion = "0.9.2",
                                suppressOutput = FALSE,
-                               developerVersion = FALSE) {
-  # Install dependencies from CRAN
-  displayProgress("Installing CRAN packages", suppressOutput = suppressOutput)
-  install.packages(.cranPackages,
-    dependencies = TRUE
-  )
-
+                               developerVersion = FALSE,
+                               lib = NULL,
+                               ...) {
+  install.packages("pkgbuild", lib = lib)
   displayProgress("Checking RTOOLS", suppressOutput = suppressOutput)
-  if (Sys.which("make") == "") { # rtools is not found
-    if (!is.null(rtoolsPath)) { # adding an existing installation of rtools to path
-      Sys.setenv(PATH = paste(rtoolsPath, Sys.getenv("PATH"), sep = ";"))
-    } else {
+  if (!pkgbuild::has_rtools()) { # rtools is not found
       displayProgress("Checking RTOOLS", success = FALSE, message = packageInstallationMessages$installRTools, suppressOutput = suppressOutput)
       return()
     }
-  }
-  if (Sys.which("make") == "") {
-    displayProgress("Checking RTOOLS", success = FALSE, message = packageInstallationMessages$RToolsNotFound, suppressOutput = suppressOutput)
-    return()
-  }
+
+  # Install dependencies from CRAN
+  displayProgress("Installing CRAN packages", suppressOutput = suppressOutput)
+  install.packages(.cranPackages,
+    dependencies = TRUE,
+    lib = lib
+  )
 
   displayProgress("Installing rClr", suppressOutput = suppressOutput)
   if (version$major == "4") {
-    install.packages(paste0("https://github.com/Open-Systems-Pharmacology/rClr/releases/download/v", rclrVersion, "/rClr_", rclrVersion, ".zip"), repos = NULL)
+    install.packages(paste0("https://github.com/Open-Systems-Pharmacology/rClr/releases/download/v", rclrVersion, "/rClr_", rclrVersion, ".zip"), repos = NULL,
+                     lib = lib)
```

Comment:
`...` is not passed as argument

Reply:
Did you ever run the tests for `rClr`? I would not risk it as we have no idea what there is and if it actually works :D


This is when I try to run them:

==> devtools::test()

```
ℹ Testing rClr
Error in nsenv[[f_name]](dirname(ns_path), package) : 
  rClr library .NET framework not found - looked under D:/Work/OSPS/Repos/rClr/inst/libs//x64 but not found in 
Calls: <Anonymous> ... test_files_setup_env -> <Anonymous> -> run_pkg_hook -> <Anonymous>
Execution halted

Exited with status 1.
```

Reply:
Well, installing the tests does not mean they are automatically run (and windows .zip packages do not contain any tests anyways - unfortunately).
But I mean, we can skip this part for rClr (I always run the tests for tlf, ospsuite.utils, ospsuite, ospsuite.reportingengine - but indeed not for rClr :)